app.rb

```diff
@@ -0,0 +1,176 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @cmd = {
+      '1': 'List all books',
+      '2': 'List all people',
+      '3': 'Create a person',
+      '4': 'Create a book',
+      '5': 'Create a rental',
+      '6': 'List all rentals for a given person id',
+      '7': 'Exit'
+    }
+
+    @books = []
+
+    @people = []
+  end
+
+  def display_cmd
+    @cmd.each do |index, command|
+      puts "#{index} - #{command}"
+    end
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No books created yet!'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}. Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'No person added to the system'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def list_all_people_with_index
+    @people.each_with_index do |person, index|
+      puts "#{index}. [#{person.class}] Name: #{person.name}, ID #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Do this person have parental permission? [Yes/No]: '
+    permission = gets.chomp.downcase
+
+    case permission
+    when 'yes' then parent_permission = true
+    when 'no' then parent_permission = false
+    end
+
+    @people.push(Student.new('11', age, name, parent_permission: parent_permission))
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people.push(Teacher.new(specialization, age, name))
+  end
+
+  def create_person
+    print 'Choose options for creating a person: Student(1), Teacher(2)? [Input the number]: '
+    case gets.chomp
+    when '1'
+      create_student
+
+    when '2'
+      create_teacher
+    end
+
+    puts 'Person Created Successfully'
+  end
```

Comment:
- [ ] Dear @YussufOB, Good job Implementing most of the functionalities, But I'm not able to add a person. Your app raises an exception when I try to do so.  Please check out the screen recording I have added and fix the issue before the next reviews.

- [ ] Also, because of this, I was unable to check the rest of the functionalities, Please make sure this and all the other functionalities are working according to the requirement before the next review request. 

### Exception I'm facing.
https://user-images.githubusercontent.com/69077061/218252129-8822a9ed-4da9-48c1-bed0-da886bf5bfe2.mp4